installer/pkg/cluster/checks.go

```diff
@@ -99,6 +106,45 @@ func checkContainerDRuntime(ctx context.Context, config *rest.Config, namespace
 	return res, nil
 }
 
+func checkKubernetesVersion(ctx context.Context, config *rest.Config, namespace string) ([]ValidationError, error) {
+	nodes, err := listNodesFromContext(ctx, config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Allow pre-releases in case provider appends anything to the version
+	constraint, err := semver.NewConstraint(kubeletVersionConstraint)
+	if err != nil {
+		return nil, err
+	}
+
+	var res []ValidationError
+	for _, node := range nodes {
+		kubeletVersion := node.Status.NodeInfo.KubeletVersion
```

Comment:
We should also ensure the version of the API/control plane matches what we expect (I'd actually think that more important than the kubelet version).

`kubectl version` offers an example on how to do that: https://github.com/kubernetes/kubectl/blob/94d80be6aba709add4dc20db166fa8c54d176653/pkg/cmd/version/version.go#L126